From 49799aa5029a873211dda26d311c29fac21e9d58 Mon Sep 17 00:00:00 2001
From: Russell King - ARM Linux <linux@arm.linux.org.uk>
Date: Tue, 6 Mar 2012 22:34:26 +0000
Subject: dmaengine: add private header file

Add a local private header file to contain definitions and declarations
which should only be used by DMA engine drivers.

We also fix linux/dmaengine.h to use LINUX_DMAENGINE_H to guard against
multiple inclusion.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
Tested-by: Linus Walleij <linus.walleij@linaro.org>
Reviewed-by: Linus Walleij <linus.walleij@linaro.org>
Acked-by: Jassi Brar <jassisinghbrar@gmail.com>
[imx-sdma.c & mxs-dma.c]
Tested-by: Shawn Guo <shawn.guo@linaro.org>
Signed-off-by: Vinod Koul <vinod.koul@linux.intel.com>
(cherry picked from commit d2ebfb335b0426deb1a4fb14e4e926d81ecd8235)

Conflicts:

	drivers/dma/amba-pl08x.c
	drivers/dma/at_hdmac.c
	drivers/dma/coh901318.c
	drivers/dma/dw_dmac.c
	drivers/dma/ep93xx_dma.c
	drivers/dma/fsldma.c
	drivers/dma/imx-dma.c
	drivers/dma/imx-sdma.c
	drivers/dma/intel_mid_dma.c
	drivers/dma/ioat/dma.c
	drivers/dma/ioat/dma_v2.c
	drivers/dma/iop-adma.c
	drivers/dma/ipu/ipu_idmac.c
	drivers/dma/mpc512x_dma.c
	drivers/dma/mv_xor.c
	drivers/dma/mxs-dma.c
	drivers/dma/pch_dma.c
	drivers/dma/pl330.c
	drivers/dma/ppc4xx/adma.c
	drivers/dma/shdma.c
	drivers/dma/ste_dma40.c
	drivers/dma/timb_dma.c
	drivers/dma/txx9dmac.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/dma/dmaengine.h   | 10 ++++++++++
 include/linux/dmaengine.h |  4 ++--
 2 files changed, 12 insertions(+), 2 deletions(-)
 create mode 100644 drivers/dma/dmaengine.h

diff --git a/drivers/dma/dmaengine.h b/drivers/dma/dmaengine.h
new file mode 100644
index 0000000..968570d
--- /dev/null
+++ b/drivers/dma/dmaengine.h
@@ -0,0 +1,10 @@
+/*
+ * The contents of this file are private to DMA engine drivers, and is not
+ * part of the API to be used by DMA engine users.
+ */
+#ifndef DMAENGINE_H
+#define DMAENGINE_H
+
+#include <linux/dmaengine.h>
+
+#endif
diff --git a/include/linux/dmaengine.h b/include/linux/dmaengine.h
index b8b4a84..56628b2 100644
--- a/include/linux/dmaengine.h
+++ b/include/linux/dmaengine.h
@@ -18,8 +18,8 @@
  * The full GNU General Public License is included in this distribution in the
  * file called COPYING.
  */
-#ifndef DMAENGINE_H
-#define DMAENGINE_H
+#ifndef LINUX_DMAENGINE_H
+#define LINUX_DMAENGINE_H
 
 #include <linux/device.h>
 #include <linux/uio.h>
-- 
1.7.10.2.565.gbd578b5

